app/javascript/components/tree-view/reducers/index.js

```diff
@@ -0,0 +1,53 @@
+import Tree, { ActionTypes } from '@manageiq/react-ui-components/dist/wooden-tree';
+import {
+  checkAll,
+  select,
+  scrollToNode,
+  expand,
+  expandAll,
+} from './others';
+
+const actionsPrefix = (prefix) => {
+  const newObject = {};
+  Object.entries(ActionTypes).forEach(([key, obj]) => {
+    newObject[key] = prefix + obj;
+  });
+  return newObject;
+};
+
+
+export const ACTIONS = {
+  ...actionsPrefix('@@tree/'),
+  EMPTY_TREE: '@@tree/clearNodes',
+  CHECK_ALL: '@@tree/checkAll',
+  SELECT: '@@tree/selectNode',
+  SELECT_SILENT: '@@tree/selectNodeSilent',
+  SCROLL_TO_NODE: '@@tree/scrollToNode',
+  EXPAND_ALL: '@@tree/expandAll',
+  EXPAND: '@@tree/expandNode',
+};
+
+const reducers = (oncheck, onclick) => (state = {}, action) => {
+  const node = action.nodeId !== undefined ? Tree.nodeSelector(state, action.nodeId) : undefined;
+
+  switch (action.type) {
+    case ACTIONS.CHECKED_DIRECTLY:
+      window[oncheck](node.attr.key, action.value, state);
```

Comment:
This and the one below might be dangerous as it can expose of a call to any function in the global space:
```jsx
<HierarchicalTreeView oncheck="alert" ... />
```

This would open just an alert box whenever you check a node, but we have some functions in the global space that might do much more damage. You should probably test that the oncheck here is among the allowed function names before invoking it. We had this implemented in `miqTreeEventSafeEval` using a [whitelist](https://github.com/ManageIQ/manageiq-ui-classic/pull/6413/files#diff-bff4131de23ce9175827098fd726ca7aL365-L379).

cc @himdel I might be wrong here, so feel free to say that @brumik's implementation is alright. I'm just worried about the deletion of a safeguard that has been added for a reason.

Reply:
No, I agree with you @skateman, if the new code needs to do `window[oncheck]` it **must** use `miqTreeEventSafeEval` (or a similar whitelist)

Reply:
My guess is that eventually these handlers will all go away or will be replaced with a single one as we move towards react in the forms around the trees. So please @brumik create a whitelist, you can go either the old way or @himdel's.